src/infi/clickhouse_orm/fields.py

```diff
@@ -70,8 +70,9 @@ def get_sql(self, with_default=True):
         else:
             return self.db_type
 
-    def is_insertable(self):
-        return self.alias is None and self.materialized is None
+    @property
+    def readonly(self):
+        return self.alias is not None or self.materialized is not None
```

Comment:
You are right, but in this case an empty string for alias or materialized makes no sense and would fail on table creation. So perhaps there should be an assert ensuring that they are not empty strings.